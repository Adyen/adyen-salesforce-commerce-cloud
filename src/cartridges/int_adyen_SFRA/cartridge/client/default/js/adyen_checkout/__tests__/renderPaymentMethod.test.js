/**
 * @jest-environment jsdom
 */
const { renderPaymentMethod } = require('../renderPaymentMethod');
const store = require('../../../../../store');

let mount;
describe('Render Payment Method', () => {
  beforeEach(() => {
    document.body.innerHTML = `
      <ul id="paymentMethodsList"></ul>
    `;
    store.componentsObj = {
      mocked_id: { node: { mocked_node: { foo: 'bar' } } },
    };

    mount = jest.fn();
    store.checkout = {
      create: jest.fn(() => ({ mount })),
    };
  });
  it('should render stored payment method with missing shopper information fields', () => {
    const paymentMethod = {
      id: 'mocked_id',
      brand: 'mocked_brand',
      name: 'mocked_name',
      lastFour: '1234',
    };
    renderPaymentMethod(
        paymentMethod,
        true,
        '/mocked_path/',
        'mocked_description',
    );
    expect(mount).toBeCalledTimes(1);
    expect(document.getElementById('paymentMethodsList')).toMatchSnapshot();
    expect(store.componentsObj).toMatchSnapshot();
  });

  it('should render payment method with shopper information fields', () => {
    document.body.innerHTML += `
    <input id="shippingFirstNamedefault" value="shippingFirstNamedefaultMock" />
      <input id="shippingLastNamedefault" value="shippingLastNamedefaultMock" />
      <input id="shippingPhoneNumberdefault" value="shippingPhoneNumberdefaultMock" />
      <input id="shippingAddressCitydefault" value="shippingAddressCitydefaultMock" />
      <input id="shippingZipCodedefault" value="shippingZipCodedefaultMock" />
      <input id="shippingCountrydefault" value="shippingCountrydefaultMock" />
      <input id="shippingStatedefault" value="shippingStatedefaultMock" />
      <input id="shippingAddressOnedefault" value="shippingAddressOnedefaultMock" />
      <input id="shippingAddressTwodefault" value="shippingAddressTwodefaultMock" />
            
      <input id="billingAddressCity" value="billingAddressCityMock" />
      <input id="billingZipCode" value="billingZipCodeMock" />
      <input id="billingCountry" value="billingCountryMock" />
      <input id="billingState" value="billingStateMock" />
      <input id="billingAddressOne" value="billingAddressOneMock" />
      <input id="billingAddressTwo" value="billingAddressTwoMock" />
      
      <span class="customer-summary-email">test@user.com</span>
    `
    const paymentMethod = {
      type: 'scheme',
      name: 'mocked_name',
      lastFour: '1234',
    };
    renderPaymentMethod(
        paymentMethod,
        false,
        '/mocked_path/',
        'mocked_description',
    );
    expect(document.getElementById('paymentMethodsList')).toMatchSnapshot();
    expect(store.componentsObj).toMatchSnapshot();
    expect(store.checkout.create.mock.calls[0][1]).toEqual({
      data: {
        personalDetails: {
          firstName: 'shippingFirstNamedefaultMock',
          lastName: 'shippingLastNamedefaultMock',
          telephoneNumber: 'shippingPhoneNumberdefaultMock',
          shopperEmail: 'test@user.com',
        },
        firstName: 'shippingFirstNamedefaultMock',
        lastName: 'shippingLastNamedefaultMock',
        telephoneNumber: 'shippingPhoneNumberdefaultMock',
        shopperEmail: 'test@user.com',
      },
      visibility: {
        personalDetails: 'editable',
        billingAddress: 'hidden',
        deliveryAddress: 'hidden',
      },
    });
  });

  it('should handle input onChange for paypal', () => {
    document.body.innerHTML += `
      <button value="submit-payment"></button>
      <div id="component_paypal"></div>
    `;
    const paymentMethod = {
      type: 'paypal',
      name: 'mocked_name',
      lastFour: '1234',
    };
    renderPaymentMethod(
        paymentMethod,
        false,
        '/mocked_path/',
        'mocked_description',
    );
    const input = document.getElementById('rb_paypal');
    input.onchange({ target: { value: 'paypal' } });
    expect(
        document.querySelector('button[value="submit-payment"]').disabled,
    ).toBeTruthy();
    expect(store.selectedMethod).toBe('paypal');
  });
});
