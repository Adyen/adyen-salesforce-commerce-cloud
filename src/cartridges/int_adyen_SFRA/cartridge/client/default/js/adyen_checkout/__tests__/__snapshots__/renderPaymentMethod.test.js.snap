// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Payment Method should render payment method with shopper information fields 1`] = `
<ul
  id="paymentMethodsList"
>
  <li
    class="paymentMethod"
  >
    
    
    <input
      id="rb_scheme"
      name="brandCode"
      type="radio"
      value="scheme"
    />
    
    
    <img
      class="paymentMethod_img"
      src="/mocked_path/card.png"
    />
    
    
    <label
      for="rb_scheme"
      id="lb_scheme"
    >
      mocked_name
    </label>
    
  
    <p>
      mocked_description
    </p>
    <div
      class="additionalFields"
      id="component_scheme"
      style="display:none"
    />
  </li>
</ul>
`;

exports[`Render Payment Method should render payment method with shopper information fields 2`] = `
Object {
  "mocked_id": Object {
    "node": Object {
      "mocked_node": Object {
        "foo": "bar",
      },
    },
  },
  "scheme": Object {
    "node": Object {
      "mount": [MockFunction] {
        "calls": Array [
          Array [
            <div
              class="additionalFields"
              id="component_scheme"
              style="display:none"
            />,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
  },
}
`;

exports[`Render Payment Method should render stored payment method with missing shopper information fields 1`] = `
<ul
  id="paymentMethodsList"
>
  <li
    class="paymentMethod"
  >
    
    
    <input
      id="rb_mocked_brand_storedCard_mocked_id"
      name="brandCode"
      type="radio"
      value="mocked_brand_storedCard_mocked_id"
    />
    
    
    <img
      class="paymentMethod_img"
      src="/mocked_path/mocked_brand.png"
    />
    
    
    <label
      for="rb_mocked_brand_storedCard_mocked_id"
      id="lb_mocked_brand_storedCard_mocked_id"
    >
      mocked_name ************1234
    </label>
    
  
    <p>
      mocked_description
    </p>
    <div
      class="additionalFields"
      id="component_mocked_brand_storedCard_mocked_id"
      style="display:none"
    />
  </li>
</ul>
`;

exports[`Render Payment Method should render stored payment method with missing shopper information fields 2`] = `
Object {
  "mocked_brand_storedCard_mocked_id": Object {
    "isValid": true,
    "node": Object {
      "mount": [MockFunction] {
        "calls": Array [
          Array [
            <div
              class="additionalFields"
              id="component_mocked_brand_storedCard_mocked_id"
              style="display:none"
            />,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
  },
  "mocked_id": Object {
    "node": Object {
      "mocked_node": Object {
        "foo": "bar",
      },
    },
  },
}
`;
